src/ustreamer/device.c

```diff
@@ -359,16 +360,20 @@ int us_device_grab_buffer(us_device_s *dev, us_hw_buffer_s **hw) {
 			}
 
 #			undef GRABBED
+#			undef FRAME_DATA
 
 			memcpy(&buf, &new, sizeof(struct v4l2_buffer));
 			buf_got = true;
 
 		} else {
-			if (buf_got && errno == EAGAIN) {
-				if (broken) {
+			if(errno == EAGAIN) {
```

Comment:
Why you changed it?

Reply:
Without this change, if `EAGAIN` is seen before a valid frame then `buf_got == false` so it will treat the error as fatal, return `-1`, and exit with `Resource temporarily unavailable`.

With this change, `EAGAIN` is never treated as fatal. Instead, there are three possible cases when `VIDIOC_DQBUF` returns `EAGAIN`:

1. If the last frame seen was broken, return `-2` (broken frame).
2. Else, if we have a valid frame, break and return `0` (success).
3. Else, `continue` to the top of the loop and try again.

I think this matches the *intent* of the original code.

Reply:
Yes, it seems you're right. I will accept it in the coming days and make a release. Thank you.